src/overlays/actors/ovl_En_Bombf/z_en_bombf.c

```diff
@@ -34,53 +35,470 @@ const ActorInit En_Bombf_InitVars = {
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        this->unk_204 = 0.0f;
+        EnBombf_SetupAction(this, func_808AEF68);
+        this->actor.room = -1;
+        return;
+    }
+
+    this->unk_204 = 1.0f;
+    if (!(this->actor.bgCheckFlags & 1)) {
+        Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 1.0f, 0.025f, 0.0f);
+        return;
+    }
+
+    Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 1.0f, 1.5f, 0.0f);
+    if (this->actor.bgCheckFlags & 2) {
+        func_800B8EF4(globalCtx, &this->actor);
+        if (this->actor.velocity.y < -6.0f) {
+            this->actor.velocity.y *= -0.3f;
+            this->actor.bgCheckFlags &= ~1;
+        }
+    } else if (this->timer >= 4) {
+        Actor_LiftActor(&this->actor, globalCtx);
+    }
+}
+
+void func_808AEF68(EnBombf* this, GlobalContext* globalCtx) {
+    if (Actor_HasNoParent(&this->actor, globalCtx)) {
+        EnBombf_SetupAction(this, func_808AEE3C);
+        this->actor.bgCheckFlags &= ~1;
+        func_808AEE3C(this, globalCtx);
+    } else {
+        this->actor.velocity.y = 0.0f;
+    }
+}
+
+void func_808AEFD4(EnBombf* this, GlobalContext* globalCtx) {
+    if (this->colliderJntSph.elements->dim.modelSphere.radius == 0) {
+        this->actor.flags |= ACTOR_FLAG_20;
+        func_8013ECE0(this->actor.xzDistToPlayer, 255, 20, 150);
+    }
+
+    this->colliderJntSph.elements->dim.modelSphere.radius = 100;
+    this->colliderJntSph.elements->dim.worldSphere.radius = 100;
+
+    if (ENBOMBF_GET(&this->actor) == ENBOMBF_1) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colliderJntSph.base);
+    }
+
+    if (globalCtx->envCtx.unk_8C.diffuseColor1[0] != 0) {
+        globalCtx->envCtx.unk_8C.diffuseColor1[0] -= 25;
+    }
+
+    if (globalCtx->envCtx.unk_8C.diffuseColor1[1] != 0) {
+        globalCtx->envCtx.unk_8C.diffuseColor1[1] -= 25;
+    }
+
+    if (globalCtx->envCtx.unk_8C.diffuseColor1[2] != 0) {
+        globalCtx->envCtx.unk_8C.diffuseColor1[2] -= 25;
+    }
+
+    if (globalCtx->envCtx.unk_8C.ambientColor[0] != 0) {
+        globalCtx->envCtx.unk_8C.ambientColor[0] -= 25;
+    }
+
+    if (globalCtx->envCtx.unk_8C.ambientColor[1] != 0) {
+        globalCtx->envCtx.unk_8C.ambientColor[1] -= 25;
+    }
+
+    if (globalCtx->envCtx.unk_8C.ambientColor[2] != 0) {
+        globalCtx->envCtx.unk_8C.ambientColor[2] -= 25;
+    }
+
+    if (this->timer == 0) {
+        Player* player = GET_PLAYER(globalCtx);
+
+        if ((player->stateFlags1 & 0x800) && (&this->actor == player->heldActor)) {
+            player->actor.child = NULL;
+            player->heldActor = NULL;
+            player->interactRangeActor = NULL;
+            player->stateFlags1 &= ~0x800;
+        }
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void EnBombf_Update(Actor* thisx, GlobalContext* globalCtx) {
+    Vec3f sp8C = D_808AFB80;
+    Vec3f sp80 = D_808AFB8C;
+    Vec3f sp74 = D_808AFB98;
+    Vec3f sp68;
+    Vec3f sp5C = D_808AFBA4;
```

Comment:
Can these assignments just be in-function static?

Reply:
Yes, but we have to put everything in the map...

Reply:
Make them in-function static.

Reply:
Actually, they're on the stack, so not static?